### DISCUSSION TITLE
Alexa / Echo Dot Gen3 not finding lamps

### DISCUSSION THREAD

**Discussion by jenswei** [hass-emulated-hue/core#319] [Q&A]
Dear Community, 
Please help getting my ZHA lights of home assistant into hass-emulated-hue.
Current symptom: the alexa echo gen3 device only finds a dummy light which was added using the template dummy_light. All real lights which use the "ZHA" integration get ignored by Alexa. However they appear and work fine within the "Hue Essentials" app.
The old "emulated_hue" integration is disabled in configuration.yaml.

Things I tried without success:
- I checked discussion threads, issues and pull requests for similar issues, it seems no one else has my exact issue.
- I tried the hass-emulated-hue "dev" addon instead of the stable one.
- I deleted all deprecated lamps within alexa app and tried to rediscover.
- I resetted the echo device by holding action button for 25 seconds.
- I deleted the emulated_hue.json and tried every approach above again
- I checked the logs for errors but just found successful connections which looked good to me (for someone who is new to hass-emulated-hue)

My goal is to replace the old emulated_hue with hass-emulated-hue, to use alexa for controlling my zha lights and also being able to use the apps like "Hue Essentials".

Any ideas what I am doing wrong? Which logs would you be interested in for troubleshooting?

**Comment by alexyao2015** [comment]
Can you go into the logs and visit whatever endpoint the Alexa is checking and send over the response when you visit it using your browser? Ensure that the template light is still present.

**Reply by jenswei** [reply]
Thanks alexyao2015,
Here is the log, verbose setting, using a fresh install of hass-emulated-hue stable version 0.2.13 with cleared config folder and previoulsy resetted echo device:
https://pastebin.com/APVFQZ49

The emulated_hue.json then looks like this:
https://pastebin.com/pSEPLH2v

All those listed lamps and groups will get found by Hue Essentials, but not by the Echo device.

**Reply by alexyao2015** [reply]
Can you paste this "/api/5c27effE00cDd5EFAfcdF018d4d3be699dae0daa/lights"

**Reply by jenswei** [reply]
Here's the result: https://pastebin.com/MzN2XBVM

**Reply by jenswei** [reply]
Could possibly the provided lamp manufacturername, productname or modelid be the culprit? I noticed the dummy light gets shown as being a signify lamp, while the others are shown as Silicon Labs EZSP and others. Perhaps the echo device rejects unknown manufaturers or lamps? Just a guess.
Edit: no, doesn't seem to be the cause since I also have original hue lamps like "manufacturername": "Philips", "productname": "Philips LTE002", "modelid": "LTE002". So these are certainly not unknown to the echo

**Reply by alexyao2015** [reply]
That is peculiar. Maybe try this build that uses the "Philips" manufacturer and see if that works?

If you download the dev addon and enter `originalmodel` into the tag field, it should let you test it.

**Reply by jenswei** [reply]
Thanks Alex for providing the originalmodel version. It didn't help finding the other lamps unfortunately. 
Here is the current log: https://pastebin.com/Q3aqEYsA
And the current output of /api/nV3NYH-KpNu3EAbcBrrW7MpMFI17gQwc8E0lIW6P/lights: https://pastebin.com/QnRUjPXL

**Reply by jenswei** [reply]
Perhaps the alexa device has issues with light no. 2 and then does not progress though the remaining file. Light no.2 in this case is a zigbee group within the zha integration.

**Reply by alexyao2015** [reply]
To test that theory, you could try disabling that in home assistant.

**Reply by jenswei** [reply]
I don't know of an easy way to disable the groups, but I stopped the addon, then edited emulated_hue.json and deleted begining from lamp 2 all zigbee group lamps until the first "real" lamp.
After starting the addon again and starting the search cycle at echo, the zigbee group lamps get added to the file again but with a higher lamp number. 
So no.2 to 6 remain empty and no. 7 should be found after no.1 (the dummy). But it didn't. Still the same result.

I really wonder if no one else has my experience with the echo device, I don't think I have a special setup in any way.

**Reply by alexyao2015** [reply]
You can disable the integration directly in home assistant. Another thing you can try is moving the dummy light id to the end to see if that makes a difference.

**Reply by jenswei** [reply]
Disabling the integration would cause all lights being gone, because all my lights are zigbee within the zha integration. I try moving the dummy light position.

**Reply by alexyao2015** [reply]
You can also try selecting the entity, choose advanced settings, then disable.

**Reply by jenswei** [reply]
Thanks. Group entities are now disabled.
If I set a lamp ID to the one which the dummy lamp had before I can successfully control this lamp using the alexa app.
However a second search only finds another dummy lamp (now with the different ID, in this case ID 17) and no other lamps.

**Reply by alexyao2015** [reply]
I see that's helpful. Can you post the new output from the api call at /lights?

**Reply by jenswei** [reply]
Yes, so the log after re-running a search is here: https://pastebin.com/7g7WBHzM.
The corresponding /lights output is https://pastebin.com/kJG7c8PN

I'm really confused why and how this echo device distinguishes between ID no.17 and e.g. no.9. There seems absolutely no difference in characteristics. Somehow it definitely does, else it would not have found the dummy at ID 17, correct?

**Reply by jenswei** [reply]
Is there anything else I can try or test? @alexyao2015

**Reply by jenswei** [reply]
Any new ideas which I could try?
